nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -113,13 +113,21 @@ def _get_spheres_rows(seeds, target_affine, target_shape, radius, allow_overlap)
     '''
     seeds = list(seeds)
 
-    # Compute world coordinates of all voxels. The mask will be applied at 
-    # the last step.
-    brain_coords = list(np.ndindex(*target_shape))
+    # Compute world coordinates of the mask and of all coordinates
+    mask = _safe_get_data(mask_img)
```

Comment:
I am not sure if I should do more tests on the `mask_img` (as in the transformation)?

Reply:
Logically, you perform any test in the high-level function, not here.